seatunnel-formats/seatunnel-format-protobuf/src/main/java/org/apache/seatunnel/format/protobuf/CompileDescriptor.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.format.protobuf;
+
+import org.apache.seatunnel.common.utils.FileUtils;
+import org.apache.seatunnel.format.protobuf.exception.ProtobufFormatErrorCode;
+import org.apache.seatunnel.format.protobuf.exception.SeaTunnelProtobufFormatException;
+
+import com.github.os72.protocjar.Protoc;
+import com.google.protobuf.DescriptorProtos;
+import com.google.protobuf.Descriptors;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.List;
+
+public class CompileDescriptor {
+
+    public static Descriptors.Descriptor compileDescriptorTempFile(
+            String protoContent, String messageName)
+            throws IOException, InterruptedException, Descriptors.DescriptorValidationException {
+        // Because Protobuf can only be dynamically parsed through the descriptor file, the file
+        // needs to be compiled and generated. The following method is used here to solve the
+        // problem: generate a temporary directory and compile .proto into a descriptor temporary
+        // file. The temporary file and directory are deleted after the JVM runs.
+        File tmpDir = createTempDirectory();
+        File protoFile = createProtoFile(tmpDir, protoContent);
+        String targetDescPath = compileProtoToDescriptor(tmpDir, protoFile);
+
+        try (FileInputStream fis = new FileInputStream(targetDescPath)) {
+            DescriptorProtos.FileDescriptorSet descriptorSet =
+                    DescriptorProtos.FileDescriptorSet.parseFrom(fis);
+            Descriptors.FileDescriptor[] descriptorsArray = buildFileDescriptors(descriptorSet);
+            return descriptorsArray[0].findMessageTypeByName(messageName);
+        }
```

Comment:
can we delete the tmp file in finally block?

Reply:
> can we delete the tmp file in finally block?
```
tmpDir.deleteOnExit();
```
   It will be automatically deleted when the program exits, without the need to call the deletion method.

Reply:
But in cluster mode, the program may never exit.

Reply:
> But in cluster mode, the program may never exit.



> But in cluster mode, the program may never exit.

You're right, I will fix it. It should indeed be deleted immediately after use